src/sentry/coreapi.py

```diff
@@ -172,18 +174,30 @@ def insert_data_to_database(self, data, start_time=None,
 
         cache_timeout = 3600
         cache_key = cache_key_for_event(data)
-        default_cache.set(cache_key, data, cache_timeout)
 
         # Attachments will be empty or None if the "event-attachments" feature
         # is turned off. For native crash reports it will still contain the
         # crash dump (e.g. minidump) so we can load it during processing.
         if attachments is not None:
             attachment_cache.set(cache_key, attachments, cache_timeout)
 
-        task = from_reprocessing and \
-            preprocess_event_from_reprocessing or preprocess_event
-        task.delay(cache_key=cache_key, start_time=start_time,
-                   event_id=data['event_id'])
+        if features.has('projects:kafka-preprocess', project=project):
```

Comment:
In order to check this feature I need a real `Project` object, which means a number of new `project = Project.objects.get_from_cache(id=self.project_id)` (seen below). Not sure if this is a problem in our ingest path or not -- and not sure of a good alternative if it is a problem.

Reply:
I guess we need a JT or Matt 👍 on this? I don't know if it will hurt anything.

Reply:
It should be fine. As long as they all are get_from_cache. In the hottest paths though, we most likely have a full Project object. So going to assume you’ve already assessed this fully.